chapters/en/chapter6/pre-trained_models.mdx

```diff
@@ -0,0 +1,326 @@
+In theory, you can use any vocoder that works on 80-bin mel spectrograms. Conveniently, 🤗 Transformers offers a vocoder 
+based on HiFi-GAN. Its weights were kindly provided by the original authors of SpeechT5.
+
+<Tip>
+
+[HiFi-GAN](https://arxiv.org/pdf/2010.05646v2.pdf) is a state-of-the-art generative adversarial network (GAN) designed 
+for high-fidelity speech synthesis. It is capable of generating high-quality and realistic audio waveforms.
+
+On the high level, HiFi-GAN consists of one generator and two discriminators. The generator is a fully convolutional 
+neural network that takes a mel-spectrogram as input and learns to produce raw audio waveforms. The discriminators' 
+role is to distinguish between real and generated audio. The two discriminators focus on different aspects of the audio. 
+
+HiFi-GAN is trained on a large dataset of high-quality audio recordings. It uses a so-called called adversarial training, 
+where the generator and discriminator networks compete against each other. Initially, the generator produces low-quality 
+audio, and the discriminator can easily differentiate it from real audio. As training progresses, the generator improves 
+its output, aiming to fool the discriminator. The discriminator, in turn, becomes more accurate in distinguishing real 
+and generated audio. This adversarial feedback loop helps both networks improve over time. Ultimately, HiFi-GAN learns to 
+generate high-fidelity audio that closely resembles the characteristics of the training data.
+
+</Tip>
+
+Loading the vocoder is as easy as any other 🤗 Transformers model.
+
+```python
+from transformers import SpeechT5HifiGan
+
+vocoder = SpeechT5HifiGan.from_pretrained("microsoft/speecht5_hifigan")
+```
+
+Now all you need to do is pass it as an argument when generating speech, and the outputs will be automatically converted to the speech waveform.
+
+```python
+speech = model.generate_speech(inputs["input_ids"], speaker_embeddings, vocoder=vocoder)
+```
+
+Let's listen to the result. The sample rate used by SpeechT5 is always 16 kHz.
+
+```python
+from IPython.display import Audio
+
+Audio(speech, rate=16000)
+```
+
+Neat! 
+
+Feel free to play with the SpeechT5 text-to-speech demo, explore other voices, experiment with inputs. Note that this 
+pre-trained checkpoint only supports English language:
+
+<iframe
+	src="https://matthijs-speecht5-tts-demo.hf.space"
+	frameborder="0"
+	width="850"
+	height="450">
+</iframe>
+
+### Massive Multilingual Speech (MMS)
+
+What if you are looking for a pre-trained model in a language other than English? Massive Multilingual Speech (MMS) is 
+another model that covers an array of speech tasks, however, it supports a large number of languages. For instance, it can 
+synthesize speech in over 1,100 languages.
+
+At the core of MMS is wav2vec2, a CTC model. If you don't remember what a Connectionist Temporal Classification (CTC) approach is, 
```

Comment:
Let us remove this line. MMS ASR is a Wav2Vec2 Model and TTS is VITS model. Nothing to do with TTS. All the ~1100 model were trained from scratch on VITS backbone for MMS dataset. They don't use any adapter layers, just a data byte-level data preprocessing and then train VITS models on each language. So we can also remove the adapter training bit from here too.

Section 7.1 - https://arxiv.org/pdf/2305.13516.pdf